jobs/icon.py

```diff
@@ -32,58 +32,35 @@ def main(cfg, model_cfg):
     ----------	
     cfg : Config
         Object holding all user-configuration parameters as attributes.
-    model_cfg : dict
-        Model configuration settings loaded from the ``config/models.yaml`` file.
     """
-    cfg = prepare_data.set_cfg_variables(cfg, model_cfg)
-
-    logfile = os.path.join(cfg.log_working_dir, "icon")
-    logfile_finish = os.path.join(cfg.log_finished_dir, "icon")
+    prepare_icon.set_cfg_variables(cfg)
+    tools.change_logfile(cfg.logfile)
 
     logging.info("Setup the namelist for an ICON run and "
                  "submit the job to the queue")
 
     # Copy icon executable
     execname = 'icon.exe'
```

Comment:
minor: `execname` variable not needed

Reply:
It is: https://github.com/C2SM/processing-chain/blob/async/cases/icon-test/icon_runjob.cfg#L345
But I will at least make a config variable out of it, like in the `cosmo.py` job.

Reply:
I don't think this variable was used in `icon.py` other than in this statement `tools.copy_file(cfg.icon_binary_file, cfg.icon_work / execname)` but I let you do whatever is fine there.